### PR TITLE
Add framework specific DgsNoOpPreparsedDocumentProvider

### PR DESCRIPTION
**Pull request by jord1e**
Pull request checklist
----

- [x] Please read our [contributor guide](https://github.com/Netflix/dgs-framework/blob/master/CONTRIBUTING.md)
- [x] Consider creating a discussion on the [discussion forum](https://github.com/Netflix/dgs-framework/discussions)
  first
- [x] Make sure the PR doesn't introduce backward compatibility issues
- [x] Make sure to have sufficient test cases

Pull Request type
----

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe):

Changes in this PR
----

Replaces the graphql-java [`NoOpPreparsedDocumentProvider`](https://github.com/graphql-java/graphql-java/blob/d783ec7ee30369ca2fdecdfd0b74e8bf6d96baa8/src/main/java/graphql/execution/preparsed/NoOpPreparsedDocumentProvider.java) with a Kotlin `object DgsNoOpPreparsedDocumentProvider` because the graphql-java one is annotated with [@Internal](https://github.com/graphql-java/graphql-java/blob/8530366f24ba316075a63402473cb2a38ca36ab3/src/main/java/graphql/Internal.java):

> ```java
> /**
>  * This represents code that the graphql-java project considers internal code that MAY not be stable within
>  * major releases.
>  *
>  * In general unnecessary changes will be avoided but you should not depend on internal classes being stable
>  */
> ```

Test adopted from https://github.com/graphql-java/graphql-java/blob/8530366f24ba316075a63402473cb2a38ca36ab3/src/test/groovy/graphql/execution/preparsed/NoOpPreparsedDocumentProviderTest.groovy

Alternatives considered
----

N/A

### CODE REVIEW

File: graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DgsNoOpPreparsedDocumentProvider.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+package com.netflix.graphql.dgs.internal
+
+import graphql.ExecutionInput
+import graphql.execution.preparsed.PreparsedDocumentEntry
+import graphql.execution.preparsed.PreparsedDocumentProvider
+import java.util.function.Function
+
+object DgsNoOpPreparsedDocumentProvider : PreparsedDocumentProvider {
```
Comment (berngp) [inline graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DgsNoOpPreparsedDocumentProvider.kt:24]: `internal object` ?

File: graphql-dgs-reactive/src/main/kotlin/com/netflix/graphql/dgs/reactive/internal/DefaultDgsReactiveQueryExecutor.kt [hunk 2]
```diff
@@ -22,16 +22,12 @@ import com.jayway.jsonpath.TypeRef
 import com.jayway.jsonpath.spi.mapper.MappingException
 import com.netflix.graphql.dgs.exceptions.DgsQueryExecutionDataExtractionException
 import com.netflix.graphql.dgs.exceptions.QueryException
-import com.netflix.graphql.dgs.internal.BaseDgsQueryExecutor
-import com.netflix.graphql.dgs.internal.DefaultDgsQueryExecutor
-import com.netflix.graphql.dgs.internal.DgsDataLoaderProvider
-import com.netflix.graphql.dgs.internal.DgsSchemaProvider
+import com.netflix.graphql.dgs.internal.*
```
Comment (berngp) [inline graphql-dgs-reactive/src/main/kotlin/com/netflix/graphql/dgs/reactive/internal/DefaultDgsReactiveQueryExecutor.kt:25]: Please keep the imports uncollapsed.

Comment (jord1e) [reply graphql-dgs-reactive/src/main/kotlin/com/netflix/graphql/dgs/reactive/internal/DefaultDgsReactiveQueryExecutor.kt]: Fixed, forgot to edit my IDEA settings.

**Comment by paulbakker** [comment]
Thanks @jord1e !

**Pull request merged**